### ISSUE TITLE
False positive detecting OctalPermissionsRule

### ISSUE THREAD

**Issue by cblakkan** [willthames/ansible-lint#109]
When trying to setgid permissions

```
[ANSIBLE0008] Octal file permissions must contain leading zero
mode: 02775
```

Looks like the regex requires exactly 3 digits, which is not always correct.

```
# At least an indent, "mode:", optional whitespace, any digits, EOL
mode_regex = re.compile(r'^\s+mode:\s*[0-9]+\s*$')
# Same as above, but with a leading zero before three digits
valid_mode_regex = re.compile(r'^\s+mode:\s*0[0-7]{3}\s*$')
```

**Comment by willthames** [comment]
@siddharthist are you able to fix this up?

**Comment by siddharthist** [comment]
@willthames I can definitely work on this! 

@cblakkan Can you explain this a little more, please? Is the fix just to require 3 or 4 digits? Why are four digits valid?

**Comment by cblakkan** [comment]
Pulled from manual pages for chmod. The first three rows illustrate the use for a 4th digit. 

```
Modes may be absolute or symbolic.  An absolute mode is an octal number constructed from the sum of one or more of the following values:

           4000    (the set-user-ID-on-execution bit) Executable files with this bit set will run with effective uid set to the uid of the file owner.  Directo-
                   ries with the set-user-id bit set will force all files and sub-directories created in them to be owned by the directory owner and not by the
                   uid of the creating process, if the underlying file system supports this feature: see chmod(2) and the suiddir option to mount(8).
           2000    (the set-group-ID-on-execution bit) Executable files with this bit set will run with effective gid set to the gid of the file owner.
           1000    (the sticky bit) See chmod(2) and sticky(8).
           0400    Allow read by owner.
           0200    Allow write by owner.
           0100    For files, allow execution by owner.  For directories, allow the owner to search in the directory.
           0040    Allow read by group members.
           0020    Allow write by group members.
           0010    For files, allow execution by group members.  For directories, allow group members to search in the directory.
           0004    Allow read by others.
           0002    Allow write by others.
           0001    For files, allow execution by others.  For directories allow others to search in the directory.
```

I believe this is the correct fix

```
valid_mode_regex = re.compile(r'^\s+mode:\s*0[0-7]{3,4}\s*$')
```

**Comment by siddharthist** [comment]
@cblakkan Thanks, I'll fix the regexp and add a test case!

**Comment by willthames** [comment]
Version 2.3.8 uploaded to pypi
